### PR TITLE
DOMA-4244 Auto isApproved

### PR DESCRIPTION
**Pull request by toplenboren**
Automatically set isApproved to true if organization and recipient has same `tin`

### CODE REVIEW

**Review by dkoviazin — APPROVED** [review]

File: apps/condo/domains/billing/schema/BillingReceipt.js [hunk 1]
```diff
@@ -152,6 +152,12 @@ const BillingReceipt = new GQLListSchema('BillingReceipt', {
             const newItem = { ...existingItem, ...resolvedData }
             const contextId = get(newItem, 'context')
             const recipient = get(newItem, 'recipient')
+            const billingIntegrationContext = await getById('BillingIntegrationOrganizationContext', contextId)
+            const billingIntegration = await getById('BillingIntegration', get(billingIntegrationContext, 'integration'))
+            const organization = await getById('Organization', get(billingIntegrationContext, 'organization'))
+            const isTrustedBankAccountSource = get(billingIntegration, 'isTrustedBankAccountSource')
+
+            const tinMatches = recipient.tin && ( recipient.tin === organization.tin )
```
Comment (sitozzz) [inline apps/condo/domains/billing/schema/BillingReceipt.js:160]:
```suggestion
            const tinMatches = recipient.tin && recipient.tin === organization.tin
```

File: apps/condo/domains/billing/schema/BillingReceipt.test.js [hunk 2]
```diff
@@ -813,6 +813,42 @@ describe('BillingReceipt', () => {
                 expect(receipt).toHaveProperty(['recipient', 'tin'])
                 expect(receipt).toHaveProperty(['receiver', 'tin'], receipt.recipient.tin)
             })
+            test('Should create recipient and set receiver.isApproved field automatically if billing integration is trusted and tin matches', async () => {
+                const { context: trustedBillingContext, organization: trustedOrganization } = await makeContextWithOrganizationAndIntegrationAsAdmin( { isTrustedBankAccountSource: true })
+                const tin = trustedOrganization.tin
+                const [trustedProperty] = await createTestBillingProperty(admin, trustedBillingContext)
+                const [trustedAccount] = await createTestBillingAccount(admin, trustedBillingContext, trustedProperty)
+                const [receipt, attrs] = await createTestBillingReceipt(admin, trustedBillingContext, trustedProperty, trustedAccount, { recipient: createTestRecipient({ tin: tin }) })
+                expect(attrs).not.toHaveProperty('receiver')
```
Comment (sitozzz) [inline apps/condo/domains/billing/schema/BillingReceipt.test.js:822]: We use to separate "prepare" and "check" blocks of code with newline
```suggestion

                expect(attrs).not.toHaveProperty('receiver')
```

**Review by sitozzz — APPROVED** [review]

File: apps/condo/domains/billing/schema/BillingReceipt.js (hunk 1 above)
Comment (toplenboren) [reply apps/condo/domains/billing/schema/BillingReceipt.js:160]: Done 👍

File: apps/condo/domains/billing/schema/BillingReceipt.js [hunk 3]
```diff
@@ -175,6 +181,7 @@ const BillingReceipt = new GQLListSchema('BillingReceipt', {
                     iec: get(recipient, 'iec'),
                     bic: get(recipient, 'bic'),
                     bankAccount: get(recipient, 'bankAccount'),
+                    isApproved: isTrustedBankAccountSource && tinMatches,
```
[the remark this answers is not in the document]
Comment (toplenboren) [reply apps/condo/domains/billing/schema/BillingReceipt.js:184]: Yep, we are going to create a script in .py

[the remark this answers is not in the document]
Comment (toplenboren) [reply apps/condo/domains/billing/schema/BillingReceipt.js:184]: Dont see the point in including the script in repo

File: apps/condo/domains/billing/schema/BillingReceipt.test.js (hunk 2 above)
Comment (toplenboren) [reply apps/condo/domains/billing/schema/BillingReceipt.test.js:822]: Done

**Review by pahaz — APPROVED** [review]

**Pull request merged**